screens/RandomQuote.js

```diff
@@ -0,0 +1,123 @@
+import React, { useState, useEffect } from 'react';
+import {
+  View,
+  Text,
+  Vibration,
+  StyleSheet,
+  ImageBackground,
+} from 'react-native';
+import { Accelerometer } from 'expo-sensors';
+
+import { QuoteLoader } from '../components/QuoteLoader';
+
+const image = {
+  url: 'https://freerangestock.com/thumbnail/38788/vector-lightbulb-ideas.jpg',
+};
+
+export const RandomQuote = () => {
+  const [data, setData] = useState({
+    x: 0,
+    y: 0,
+    z: 0,
+  });
+  const [quote, setQuote] = useState({});
+  const [loading, setLoading] = useState(false);
+  const [subscription, setSubscription] = useState(null);
+
+  useEffect(() => {
+    generateQuote();
+  }, []);
+
+  useEffect(() => {
+    Accelerometer.setUpdateInterval(1000);
+    subscribe();
+    return () => unsubscribe();
+  }, []);
+
+  useEffect(() => {
+    if (isShakingEnough(data)) {
+      Vibration.vibrate();
+      generateQuote();
+    }
+  }, [data]);
+
+  const subscribe = () => {
+    setSubscription(
+      Accelerometer.addListener((accelerometerData) => {
+        setData(accelerometerData);
+      })
+    );
+  };
+
+  const unsubscribe = () => {
+    subscription && subscription.remove();
+    setSubscription(null);
+  };
+
+  const generateQuote = () => {
+    setLoading(true);
```

Comment:
Maybe we can add a nice loading indicator here, with ternary operator so the user can see that it is being loaded!